File: pkg/codegen/testing/test/testdata/assets-archives-pp/dotnet/assets-archives.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    var siteBucket = new Aws.S3.Bucket("siteBucket");

    var testFileAsset = new Aws.S3.BucketObject("testFileAsset", new()
    {
        Bucket = siteBucket.Id,
        Source = new FileAsset("file.txt"),
    });

    var testStringAsset = new Aws.S3.BucketObject("testStringAsset", new()
    {
        Bucket = siteBucket.Id,
        Source = new StringAsset("<h1>File contents</h1>"),
    });

    var testRemoteAsset = new Aws.S3.BucketObject("testRemoteAsset", new()
    {
        Bucket = siteBucket.Id,
        Source = new RemoteAsset("https://pulumi.test"),
    });

    var testFileArchive = new Aws.Lambda.Function("testFileArchive", new()
    {
        Role = siteBucket.Arn,
        Code = new FileArchive("file.tar.gz"),
    });

    var testRemoteArchive = new Aws.Lambda.Function("testRemoteArchive", new()
    {
        Role = siteBucket.Arn,
        Code = new RemoteArchive("https://pulumi.test/foo.tar.gz"),
    });

    var testAssetArchive = new Aws.Lambda.Function("testAssetArchive", new()
    {
        Role = siteBucket.Arn,
        Code = new AssetArchive(new Dictionary<string, AssetOrArchive>
        {
            ["file.txt"] = new FileAsset("file.txt"),
            ["string.txt"] = new StringAsset("<h1>File contents</h1>"),
            ["remote.txt"] = new RemoteAsset("https://pulumi.test"),
            ["file.tar"] = new FileArchive("file.tar.gz"),
            ["remote.tar"] = new RemoteArchive("https://pulumi.test/foo.tar.gz"),
            [".nestedDir"] = new AssetArchive(new Dictionary<string, AssetOrArchive>
            {
                ["file.txt"] = new FileAsset("file.txt"),
                ["string.txt"] = new StringAsset("<h1>File contents</h1>"),
                ["remote.txt"] = new RemoteAsset("https://pulumi.test"),
                ["file.tar"] = new FileArchive("file.tar.gz"),
                ["remote.tar"] = new RemoteArchive("https://pulumi.test/foo.tar.gz"),
            }),
        }),
    });

});

File: pkg/codegen/testing/utils/testdata/assets-archives-pp/python/assets-archives.py
import pulumi
import pulumi_aws as aws

site_bucket = aws.s3.Bucket("siteBucket")
test_file_asset = aws.s3.BucketObject("testFileAsset",
    bucket=site_bucket.id,
    source=pulumi.FileAsset("file.txt"))
test_string_asset = aws.s3.BucketObject("testStringAsset",
    bucket=site_bucket.id,
    source=pulumi.StringAsset("<h1>File contents</h1>"))
test_remote_asset = aws.s3.BucketObject("testRemoteAsset",
    bucket=site_bucket.id,
    source=pulumi.RemoteAsset("https://pulumi.test"))
test_file_archive = aws.lambda_.Function("testFileArchive",
    role=site_bucket.arn,
    code=pulumi.FileArchive("file.tar.gz"))
test_remote_archive = aws.lambda_.Function("testRemoteArchive",
    role=site_bucket.arn,
    code=pulumi.RemoteArchive("https://pulumi.test/foo.tar.gz"))
test_asset_archive = aws.lambda_.Function("testAssetArchive",
    role=site_bucket.arn,
    code=pulumi.AssetArchive({
        "file.txt": pulumi.FileAsset("file.txt"),
        "string.txt": pulumi.StringAsset("<h1>File contents</h1>"),
        "remote.txt": pulumi.RemoteAsset("https://pulumi.test"),
        "file.tar": pulumi.FileArchive("file.tar.gz"),
        "remote.tar": pulumi.RemoteArchive("https://pulumi.test/foo.tar.gz"),
        ".nestedDir": pulumi.AssetArchive({
            "file.txt": pulumi.FileAsset("file.txt"),
            "string.txt": pulumi.StringAsset("<h1>File contents</h1>"),
            "remote.txt": pulumi.RemoteAsset("https://pulumi.test"),
            "file.tar": pulumi.FileArchive("file.tar.gz"),
            "remote.tar": pulumi.RemoteArchive("https://pulumi.test/foo.tar.gz"),
        }),
    }))
